src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -90,15 +89,30 @@ class ActionPlanService(
 
   fun approveActionPlan(id: UUID, user: AuthUser): ActionPlan {
     val actionPlan = getActionPlan(id)
+
+    actionPlan.referral.approvedActionPlan?. let {
+      reassignSessions(it.id, actionPlan)
+    } ?: run {
+      actionPlanSessionsService.createUnscheduledSessionsForActionPlan(actionPlan)
+    }
+
     actionPlan.approvedAt = OffsetDateTime.now()
     actionPlan.approvedBy = authUserRepository.save(user)
 
-    actionPlanSessionsService.createUnscheduledSessionsForActionPlan(actionPlan)
     val approvedActionPlan = actionPlanRepository.save(actionPlan)
     actionPlanEventPublisher.actionPlanApprovedEvent(approvedActionPlan)
     return approvedActionPlan
   }
 
+  private fun reassignSessions(fromActionPlanId: UUID, toActionPlan: ActionPlan) {
```

Comment:
I am wondering if it would be better to clone rather than move it across?
This is because if they wish to view the previous action plan there will be no session tied to it?

Reply:
+1, would be better to not mutate data, that can get hairy over time

We have a few options

- copying/cloning
- removing the action plan ID -- delivery_session link, and use referral -- delivery_session link, so that same sessions are used across _all_ actions plans